### DISCUSSION TITLE
Pushing to Netdata

### DISCUSSION THREAD

**Discussion by rshen-tibco** [netdata/netdata#15874] [Q&A] [answered]
https://www.netdata.cloud/roadmap/ says that Netdata is able to collect metrics using any protocol and any methodology, using both pull (Netdata asks for metrics) and push (Netdata waits for metrics to come) architectures. But I haven't found any information on pushing metrics to Netdata online. Has anyone had experience with using this functionality, or does anyone know where there is documentation on it?

**Comment by andrewm4894** [comment, marked as answer]
I think StatsD might be the main "push" example that comes to mind: https://learn.netdata.cloud/docs/data-collection/apm/statsd

e.g. in [this example](https://learn.netdata.cloud/docs/data-collection/apm/statsd#shell) you are pushing metrics to the statsd endpoint of netdata at port 8125

now that you mention it I'm not sure we have much in our docs that discusses push vs pull which is a common thing we probably should cover more. 

do you have a specific use case in mind or is it just more of a general question?

i know we are looking to more fully support OTEL stuff soon ([prometheus collector](https://learn.netdata.cloud/docs/data-collection/apm/openmetrics-endpoints) is often enough but we need to add some more parts for OTEL in a more general sense) and am sure things like receivers and push vs pull will be a big part of that. https://opentelemetry.io/docs/collector/configuration/#receivers

**Reply by rshen-tibco** [reply]
More of a general question! I'm just starting to look at Netdata, and there wasn't much information online about what I thought seemed like it could be a pretty interesting feature. It would definitely be awesome to have more documentation on it! Thank you for the overview 👍 I'll definitely ask again if a more specific use case arises.
